cmd/sourced/compose/workdir/common.go

```diff
@@ -330,7 +239,116 @@ func Validate(dir string) error {
 	return nil
 }
 
+// path returns the absolute path to
+// $HOME/.sourced/workdirs/workdir
+func absolutePath(workdir string) (string, error) {
+	path, err := workdirsPath()
+	if err != nil {
+		return "", err
+	}
+
+	// On windows replace C:\path with C\path
+	workdir = strings.Replace(workdir, ":", "", 1)
+
+	return filepath.Join(path, workdir), nil
+}
+
 func hasContent(path, file string) bool {
 	empty, err := isEmptyFile(filepath.Join(path, file))
 	return !empty && err == nil
 }
+
+// isEmptyFile returns true if the file does not exist or if it exists but
+// contains empty text
+func isEmptyFile(path string) (bool, error) {
+	_, err := os.Stat(path)
+	if err != nil {
+		if !os.IsNotExist(err) {
+			return false, err
+		}
+
+		return true, nil
+	}
+
+	contents, err := ioutil.ReadFile(path)
+	if err != nil {
+		return false, err
+	}
+
+	strContents := string(contents)
+	return strings.TrimSpace(strContents) == "", nil
+}
+
+// common initialization for both local and remote data
+func initWorkdir(workdirPath string, makeEnvContent func() string) error {
```

Comment:
Why not `envContent envFile` insteaf of `makeEnvContent func() string`?

Reply:
hm. Makes sense to change. I added `envFile` after I wrote this function.